packages/store/addon/-private/system/core-store.ts

```diff
@@ -3344,6 +3343,13 @@ abstract class CoreStore extends Service {
     // property defined on the adapter
     let adapter = this.adapterFor(modelName);
     let serializerName = get(adapter, 'defaultSerializer');
+
+    deprecate('deprecate adapter.serializer and adapter.defaultSerializer fallbacks in store.serializerFor', !serializerName, {
```

Comment:
We can give a better description here of what is deprecated and how to resolve it